### PR TITLE
fix 191: Bag access is done via cached value

### LINKED ISSUE #191: Can drop bags with item in it when storing an equipped item
**Describe the bug**
After storing an item you had equipped into a bag, the bag can be dropped.

**Version Encountered**
0.6.2.416

**To Reproduce**
1. Equip an item.
2. Store the equipped item into a bag without unequipping it.
3. Drop the bag, and see that you can (sometimes).

It doesn't always let me do this either, but something DBhandler.GetBag seems to return null when one definitely exists, thus skipping over the check to see if the bag is full.

Also noticed that sometimes the ItemId is just not filled in on bags I'm trying to drop to the map.
Oddly enough I can't reproduce this with any other item type.. Or haven't yet at least.

**Expected behavior**
You should **NOT** be able to drop a bag containing items.

**Screenshots**
Preview

### PR DESCRIPTION
**Pull request by lodicolo**
Resolves #191

### CODE REVIEW

**Comment by lodicolo** [comment]
@jcsnider Let @Azurebeats review this before merging.

It's an informative review because the previous access was using a pitfall of the engine (accessing a field via the DB when it has a new value but hasn't been committed yet). I'm sure there are more pitfalls like this elsewhere in the engine.

**Comment by lodicolo** [comment]
Making a couple changes because I realized some method signatures were ridiculous and a proper check wasn't being done.

File: Intersect.Server/Entities/Player.cs [hunk 1]
```diff
@@ -1611,85 +1649,102 @@ public List<InventorySlot> FindOpenInventorySlots()
...
-                    PacketSender.SendChatMsg(this, Strings.Bags.dropnotempty, CustomColors.Alerts.Error);
-
-                    return;
-                }
+                PacketSender.SendChatMsg(this, Strings.Bags.dropnotempty, CustomColors.Alerts.Error);
+                return false;
             }
 
-            var map = MapInstance.Get(MapId);
+            var map = Map;
```
[the remark this answers is not in the document]
Comment (lodicolo) [reply Intersect.Server/Entities/Player.cs:1717]: `Map` is actually a wrapper of `MapInstance.Get(MapId)`, so it only appears to be locally available but it really performs a lookup on each access.

[the remark this answers is not in the document]
Comment (Azurebeats) [reply Intersect.Server/Entities/Player.cs:1717]: Ah, right. Fair enough!

**Pull request merged**